be/src/exprs/runtime_filter.cpp

```diff
@@ -735,7 +735,9 @@ class RuntimePredicateWrapper {
         _is_bloomfilter = true;
         // we won't use this class to insert or find any data
         // so any type is ok
-        _context.bloom_filter_func.reset(create_bloom_filter(PrimitiveType::TYPE_INT));
+        _context.bloom_filter_func.reset(create_bloom_filter(_column_return_type == INVALID_TYPE
+                                                                     ? PrimitiveType::TYPE_INT
```

Comment:
why assign type int when it is invalid type?

Reply:
the up comment tell the reason: 
```we won't use this class to insert or find any data
   so any type is ok
```

after change the logic, the `_column_return_type` be valid in `apply filter`, can reuse the mem of it